tests/ui/lint/non_local_definitions.rs

```diff
@@ -0,0 +1,117 @@
+// check-pass
+// edition:2021
+
+use std::fmt::{Debug, Display};
+
+struct Test;
+
+impl Debug for Test {
+    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        todo!()
+    }
+}
+
+mod do_not_lint_mod {
+    pub trait Tait {}
+
+    impl super::Test {
+        fn hugo() {}
+    }
+
+    impl Tait for super::Test {}
+}
+
+const _: () = {
+    trait Uto {}
+
+    impl Uto for Test {}
```

Comment:
Maybe add a comment that this is intentionally not linted?

Reply:
Done.